src/textual/_compose.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .app import App
+    from .widget import Widget
+
+
+def compose(node: App | Widget) -> list[Widget]:
+    """Compose child widgets.
+
+    Args:
+        node: The parent node.
+
+    Returns:
+        A list of widgets.
+    """
+    app = node.app
```

Comment:
You typed `node` as `App | Node`. If `node` is the app, can you really do `node.app`?

Reply:
That is an oddity, but App inherits from MessagePump and MessagePump has an `app` property. Mypy would inform me if that wasn't the case!